File: DataSources/Oracle/server/aspnet/RevealSdk.Server/Reveal/DataSourceProvider.cs
using Reveal.Sdk;
using Reveal.Sdk.Data;
using Reveal.Sdk.Data.Oracle;

namespace RevealSdk.Server.Reveal
{
    public class DataSourceProvider : IRVDataSourceProvider
    {
        public Task<RVDataSourceItem> ChangeDataSourceItemAsync(IRVUserContext userContext, string dashboardId,
            RVDataSourceItem dataSourceItem)
        {
            if (dataSourceItem is RVOracleDataSourceItem oracleDataSourceItem)
            {
                //update underlying data source
                ChangeDataSourceAsync(userContext, oracleDataSourceItem.DataSource);

                //only change the table if we have selected our custom data source item
                if (oracleDataSourceItem.Id == "MyOracleSIDDataSourceItem")
                {
                    oracleDataSourceItem.Table = "your-table";
                }
                
                if (oracleDataSourceItem.Id == "MyOracleServiceDataSourceItem")
                {
                    oracleDataSourceItem.Table = "your-table";
                }
            }

            return Task.FromResult(dataSourceItem);
        }

        public Task<RVDashboardDataSource> ChangeDataSourceAsync(IRVUserContext userContext,
            RVDashboardDataSource dataSource)
        {
            //using SID
            if (dataSource is RVOracleSIDDataSource oracleSidDataSource)
            {
                oracleSidDataSource.Host = "your-host";
                oracleSidDataSource.Database = "your-database";
                oracleSidDataSource.SID = "your-service-sid";
            }

            //using service name
            if (dataSource is RVOracleServiceDataSource oracleServiceDataSource)
            {
                oracleServiceDataSource.Host = "your-host";
                oracleServiceDataSource.Database = "your-database";
                oracleServiceDataSource.Service = "your-service-name";
            }

            return Task.FromResult(dataSource);
        }
    }
}

File: DataSources/Oracle/server/spring-boot-jersey/src/main/java/com/server/reveal/DataSourceProvider.java
package com.server.reveal;

import com.infragistics.reveal.sdk.api.IRVDataSourceProvider;
import com.infragistics.reveal.sdk.api.IRVUserContext;
import com.infragistics.reveal.sdk.api.model.*;

import java.util.Objects;

public class DataSourceProvider implements IRVDataSourceProvider {
    public RVDataSourceItem changeDataSourceItem(IRVUserContext userContext, String dashboardsID, RVDataSourceItem dataSourceItem) {

        if (dataSourceItem instanceof RVOracleDataSourceItem oracleDataSourceItem) {

            //update underlying data source
            changeDataSource(userContext, dataSourceItem.getDataSource());

            //only change the table if we have selected our custom data source item
            if (Objects.equals(dataSourceItem.getId(), "MyOracleSIDDataSourceItem")) {
                oracleDataSourceItem.setTable("your-table");
            }

            if (Objects.equals(dataSourceItem.getId(), "MyOracleServiceDataSourceItem")) {
                oracleDataSourceItem.setTable("your-table");
            }
        }
        return dataSourceItem;
    }

    public RVDashboardDataSource changeDataSource(IRVUserContext userContext, RVDashboardDataSource dataSource) {
        //using SID
        if (dataSource instanceof RVOracleSIDDataSource oracleSIDDataSource) {
            oracleSIDDataSource.setHost("your-host");
            oracleSIDDataSource.setDatabase("your-database");
            oracleSIDDataSource.setSID("your-sid");
        }

        //using service name
        if (dataSource instanceof RVOracleServiceDataSource oracleServiceDataSource) {
            oracleServiceDataSource.setHost("your-host");
            oracleServiceDataSource.setDatabase("your-database");
            oracleServiceDataSource.setService("your-service-name");
        }

        return dataSource;
    }
}
